### ISSUE TITLE
Rasterizing patch changes filling of hatches in pdf backend

### ISSUE THREAD

**Issue by claria** [matplotlib/matplotlib#6228]
I want to plot a patch with the '*' hatch that is filled. The hatch is always filled when using
the agg backend but in the pdf backend only when it is rasterized. 

As the legend artists do not use the rasterized prop (correct?), it is inconsistent.

mpl: 1.5.1, python 3.5.1, arch linux

Code to reproduce:

```
import matplotlib
import matplotlib.pyplot as plt

fig = plt.figure()
ax = fig.add_subplot(111)

# patch without hatching
patch1 = matplotlib.patches.Rectangle((0.5, 0.5), 1., 1., color='red', fill=False, hatch='*', label='patch1')
# patch with hatching
patch2 = matplotlib.patches.Rectangle((2.0, 0.5), 1., 1., color='blue', fill=False, hatch='*', rasterized=True, label='patch2')

ax.add_patch(patch1)
ax.add_patch(patch2)

ax.set_xlim(0., 3.5)
ax.set_ylim(0., 3)

ax.legend()


fig.savefig('asdf.png')
fig.savefig('asdf.pdf')
```

PNG output:

![asdf](https://cloud.githubusercontent.com/assets/1030408/14060097/f78fa8f6-f358-11e5-8fa8-2a5349a72fd7.png)

PDF output:

![asdf pdf](https://cloud.githubusercontent.com/assets/1030408/14060099/045d899a-f359-11e5-83b2-d44e284f4773.png)

**Comment by tacaswell** [comment]
This may be addressed by #6198

**Comment by tacaswell** [comment]
No, not fixed.

In pdf when an artist is rasterized that just means that artist is rendered via the Agg backend (the same as png) and then that is inserted into the pdf (hence why they match).  We probably should propagate the rasterized flag to the legend.  That isn't fool-proof as the artists maybe rasterized based on zorder as well, but better than nothing.

**Comment by tacaswell** [comment]
Of course doing that exposes #6227 ;)

**Comment by QuLogic** [comment]
So the legend is kind of unrelated here; it's that Agg renders with a filled hatch pattern when `fill=False` while PDF does not. Probably either one can be fixed directly rather easily, but which one is the expected behaviour here?

**Comment by tacaswell** [comment]
I would go with the Agg behavior as 'correct'.

**Comment by QuLogic** [comment]
Well that's good, because PDF's the easier one to fix.

**Comment by NelleV** [comment]
Closed in #7421
